### PR TITLE
Improve security store & remove all pending guardians after recovery

### PR DESCRIPTION
**Pull request by dong77**
Remove `cancelGuardianAddition` and `cancelGuardianRemoval`.

Please carefully review `addGuardian` and `removeGuardian` functions, I am not so confident.

### CODE REVIEW

**Review by kongliangzhong — APPROVED** [review]

File: packages/hebao_v1/contracts/stores/GuardianStore.sol [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+            _cleanExpiredGuardians(wallet);
+        } else {
+            Data.Guardian memory g = w.guardians[pos - 1];
+            bool isGuardianActive = _isActivated(g) && !_isExpired(g);
+
+            // If this wallet is active, do nothing.
+            if (!isGuardianActive) {
+                w.guardians[pos - 1].validSince = validSince.toUint40();
+                w.guardians[pos - 1].validUntil = 0;
+            }
```
Comment (Brechtpd) [inline packages/hebao_v1/contracts/stores/GuardianStore.sol:148]: If the guardian is pending for removal `isGuardianActive` will be true, but `validUntil` is not reset to 0 in that case, so the pending removal is not cancelled, which I think is the desired outcome.

`w.guardians[pos - 1].validUntil = 0;` I think should always be done.

Comment (dong77) [reply packages/hebao_v1/contracts/stores/GuardianStore.sol]: please see the new impl.

File: packages/hebao_v1/contracts/stores/GuardianStore.sol [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+
+        if (_isAdded(g)) {
+            w.guardians[pos - 1].status = uint8(Data.GuardianStatus.REMOVE);
+            w.guardians[pos - 1].timestamp = validUntil.toUint64();
+            return validUntil;
+        }
+
+        if (_isPendingAddition(g)) {
+            w.guardians[pos - 1].status = uint8(Data.GuardianStatus.REMOVE);
+            w.guardians[pos - 1].timestamp = 0;
```
Comment (kongliangzhong) [inline packages/hebao_v1/contracts/stores/GuardianStore.sol:199]: delete w.guardianIdx[addr] here.

File: packages/hebao_v1/contracts/stores/GuardianStore.sol [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+        Data.Guardian memory g = w.guardians[pos - 1];
+
+        if (_isAdded(g)) {
+            w.guardians[pos - 1].status = uint8(Data.GuardianStatus.REMOVE);
+            w.guardians[pos - 1].timestamp = validUntil.toUint64();
+            return validUntil;
+        }
+
+        if (_isPendingAddition(g)) {
+            w.guardians[pos - 1].status = uint8(Data.GuardianStatus.REMOVE);
```
Comment (kongliangzhong) [inline packages/hebao_v1/contracts/stores/GuardianStore.sol:198]: remove pending guardian immedately here.

Comment (Brechtpd) [reply packages/hebao_v1/contracts/stores/GuardianStore.sol:198]: Isn't that what happens here? It's just not removed from the list which I think is fine.

**Review by Brechtpd — APPROVED** [review]

**Pull request merged**